### PR TITLE
Docs for MessageConfigurator

### PR DESCRIPTION
**Pull request by tulioag**
Fixes: vaadin/collaboration-engine-internal#594
~WIP: Needs vaadin 21.0.0.alpha3~ Ready for review.

### CODE REVIEW

File: src/main/java/com/vaadin/demo/ce/MessageListDocumentation.java [hunk 1]
```diff
@@ -73,4 +73,27 @@ private void customSubmitter() {
         });
         // end::message-list-submitter[]
     }
+
+    private void messageConfigurator() {
```
[the remark this answers is not in the document]
Comment (tulioag) [reply src/main/java/com/vaadin/demo/ce/MessageListDocumentation.java:77]: @muse-dev ignore

File: articles/ce/collaboration-message-list.asciidoc [hunk 2]
```diff
@@ -143,3 +143,20 @@ This callback should be used to set up any listeners for submitting new messages
...
+== Customizing Messages
+`CollaborationMessageList` enables changing the properties of `MessageListItem` after they are generated.
+This is done by providing a `MessageConfigurator`.
+
+Here is an example of how to use a message configurator to add a custom style to the current user's messages.
+Refer to <<{articles}/ds/components/messages#theming, messages theming documentation>> for details on how style messages based on the theme.
+[source,java]
+----
+include::{root}/src/main/java/com/vaadin/demo/ce/MessageListDocumentation.java[tags=message-list-configurator;message-list-configurator-style;!*,indent=0]
+----
```
Comment (magi42) [inline articles/ce/collaboration-message-list.asciidoc:156]:
```suggestion
The following example shows how to use a message configurator to add a custom style to the current user's messages:

[source,java]
----
include::{root}/src/main/java/com/vaadin/demo/ce/MessageListDocumentation.java[tags=message-list-configurator;message-list-configurator-style;!*,indent=0]
----

Please refer to <<{articles}/ds/components/messages#theming, Theming>> in Messages documentation for details on how to style messages based on the theme.
```

File: articles/ce/collaboration-message-list.asciidoc [hunk 3]
```diff
@@ -143,3 +143,20 @@ This callback should be used to set up any listeners for submitting new messages
...
+
+The message configurator can also be used to change the message's text, for example censoring certain words in messages.
```
Comment (magi42) [inline articles/ce/collaboration-message-list.asciidoc:158]:
```suggestion
The message configurator can also be used to change the text content of messages, for example, to censor certain words in messages, as is done in the following example.
```

**Review by magi42 — CHANGES_REQUESTED** [review]

**Comment by tulioag** [comment]
@magi42 Is anything missing?

**Review by magi42 — APPROVED** [review]

**Comment by magi42** [comment]
At least the text looks OK, someone else should check the Java part. One minor thing there is code formatting in the two code examples. If possible, rather have
```
CollaborationMessageList collaborationMessageList =
        new CollaborationMessageList(userInfo, topicId);
```
No prob if it's difficult because of automatic formatting or something.

**Comment by tulioag** [comment]
> At least the text looks OK, someone else should check the Java part. One minor thing there is code formatting in the two code examples. If possible, rather have
> 
> ```
> CollaborationMessageList collaborationMessageList =
>         new CollaborationMessageList(userInfo, topicId);
> ```
> 
> No prob if it's difficult because of automatic formatting or something.

Fixed

**Pull request merged**